### PR TITLE
Add sfx for writing on paper

### PR DESCRIPTION
**Pull request by Krunklehorn**
## About the PR

A sound is played when a player finishes writing on paper.

## Why / Balance

Nice feature to make writing on paper more obvious. Helps Mimes get your attention.

## Technical details

`SharedPaperComponent` can now specify a sound that `PaperSystem` checks for and plays during `OnInputTextMessage()`.

Five mono ogg snippets have been supplied. Let me know if this is too many or if they require further processing.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Krunk
- add: Sound effects have been added for writing on paper.

### CODE REVIEW

**Comment by UbaserB** [comment]
very nice, can i get a preview of the sfx?

**Comment by Krunklehorn** [comment]
Yes, WAVs here...
https://discord.com/channels/310555209753690112/536955542913024015/1207551185729355796

File: Resources/Prototypes/Entities/Objects/Misc/paper.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@
       map: ["enum.PaperVisualLayers.Stamp"]
       visible: false
   - type: Paper
+    sound: !type:SoundCollectionSpecifier
+      collection: PaperScribbles
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Misc/paper.yml:19]:
```suggestion
  - type: Paper
    sound:
      collection: PaperScribbles
```

type tag isnt needed here

**Comment by JoeTheJanitor** [comment]
> Helps Mimes get your attention.
Mime stamp and bag are silent, it would actually make sense for mime crayon to be silent as well

**Comment by Krunklehorn** [comment]
That would require changes to the way paper handles it's UI state which is beyond the scope of this PR.

File: Content.Server/Paper/PaperSystem.cs [hunk 2]
```diff
@@ -160,6 +160,9 @@ private void OnInputTextMessage(EntityUid uid, PaperComponent paperComp, PaperIn
                 if (args.Session.AttachedEntity != null)
                     _adminLogger.Add(LogType.Chat, LogImpact.Low,
                         $"{ToPrettyString(args.Session.AttachedEntity.Value):player} has written on {ToPrettyString(uid):entity} the following text: {args.Text}");
+
+                if (paperComp.Sound != null)
+                    _audio.PlayPvs(paperComp.Sound, uid);
```
Comment (EmoGarbage404) [inline Content.Server/Paper/PaperSystem.cs:165]:
```suggestion
                _audio.PlayPvs(paperComp.Sound, uid);
```
don't need to check for null SoundSpecifier.

File: Content.Shared/Paper/SharedPaperComponent.cs [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
 
 namespace Content.Shared.Paper;
 
 public abstract partial class SharedPaperComponent : Component
 {
+    /// <summary>
+    /// Sound played after writing to the paper.
+    /// </summary>
+    [DataField("sound")]
+    public SoundSpecifier? Sound { get; private set; } = null;
```
Comment (EmoGarbage404) [inline Content.Shared/Paper/SharedPaperComponent.cs:12]: just give this the default writing sound since i imagine most things will want this.

**Review by EmoGarbage404 — DISMISSED** [review]

File: Content.Shared/Paper/SharedPaperComponent.cs (hunk 3 above)
Comment (Krunklehorn) [reply Content.Shared/Paper/SharedPaperComponent.cs]: It's okay to reference a prototype by string for a field member that requires a default?

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**